### ISSUE TITLE
Not able to upload file or create folder

### ISSUE THREAD

**Issue by praful-dhabekar** [alexusmai/laravel-file-manager#77]
I've followed all the instruction from [installation](https://github.com/alexusmai/laravel-file-manager/blob/master/docs/installation.md#installation). But I am unable to do anything. I can't upload file, I can't create folder. When I click on upload icon nothing happens.

Here's my **file-manager.php**

```<?php

use Alexusmai\LaravelFileManager\Services\ConfigService\DefaultConfigRepository;
use Alexusmai\LaravelFileManager\Services\ACLService\ConfigACLRepository;

return [

    /**
     * Set Config repository
     *
     * Default - DefaultConfigRepository get config from this file
     */
    'configRepository' => DefaultConfigRepository::class,

    /**
     * ACL rules repository
     *
     * Default - ConfigACLRepository (see rules in - aclRules)
     */
    'aclRepository' => ConfigACLRepository::class,

    //********* Default configuration for DefaultConfigRepository **************

    /**
     * List of disk names that you want to use
     * (from config/filesystems)
     */
    'diskList' => ['local'],

        /**
     * Default disk for left manager
     * null - auto select the first disk in the disk list
     */
    'leftDisk'  => 'local',

    /**
     * Default disk for right manager
     * null - auto select the first disk in the disk list
     */
    'rightDisk' => null,

    /**
     * Default path for left manager
     * null - root directory
     */
    'leftPath'  => null,

    /**
     * Default path for right manager
     * null - root directory
     */
    'rightPath' => null,

    /**
     * Image cache ( Intervention Image Cache )
     *
     * set null, 0 - if you don't need cache (default)
     * if you want use cache - set the number of minutes for which the value should be cached
     */
    'cache' => null,

    /**
     * File manager modules configuration
     *
     * 1 - only one file manager window
     * 2 - one file manager window with directories tree module
     * 3 - two file manager windows
     */
    'windowsConfig' => 2,

    /**
     * File upload - Max file size in KB
     *
     * null - no restrictions
     */
    'maxUploadFileSize' => 5000,

    /**
     * File upload - Allow these file types
     *
     * [] - no restrictions
     */
    'allowFileTypes' => [],

    /***************************************************************************
     * Middleware
     *
     * Add your middleware name to array -> ['web', 'auth', 'admin']
     * !!!! RESTRICT ACCESS FOR NON ADMIN USERS !!!!
     */
    'middleware' => ['web'],

    /***************************************************************************
     * ACL mechanism ON/OFF
     *
     * default - false(OFF)
     */
    'acl' => false,

    /**
     * Hide files and folders from file-manager if user doesn't have access
     *
     * ACL access level = 0
     */
    'aclHideFromFM' => true,

    /**
     * ACL strategy
     *
     * blacklist - Allow everything(access - 2 - r/w) that is not forbidden by the ACL rules list
     *
     * whitelist - Deny anything(access - 0 - deny), that not allowed by the ACL rules list
     */
    'aclStrategy' => 'blacklist',

    /**
     * ACL Rules cache
     *
     * null or value in minutes
     */
    'aclRulesCache' => null,

    //********* Default configuration for DefaultConfigRepository END **********


    /***************************************************************************
     * ACL rules list - used for default ACL repository (ConfigACLRepository)
     *
     * 1 it's user ID
     * null - for not authenticated user
     *
     * 'disk' => 'disk-name'
     *
     * 'path' => 'folder-name'
     * 'path' => 'folder1*' - select folder1, folder12, folder1/sub-folder, ...
     * 'path' => 'folder2/*' - select folder2/sub-folder,... but not select folder2 !!!
     * 'path' => 'folder-name/file-name.jpg'
     * 'path' => 'folder-name/*.jpg'
     *
     * * - wildcard
     *
     * access: 0 - deny, 1 - read, 2 - read/write
     */
    'aclRules' => [
        null => [
            //['disk' => 'public', 'path' => '/', 'access' => 2],
        ],
        1 => [
            //['disk' => 'public', 'path' => 'images/arch*.jpg', 'access' => 2],
            //['disk' => 'public', 'path' => 'files/*', 'access' => 1],
        ],
    ],
];
```

Here's my **home.blade.php**

```
@extends('layouts.app')

@section('content')
<div class="container">
    <div style="height: 700px;">
        <div id="fm"></div>
    </div>
</div>
@endsection

```
Here's my **app.blade.php**
```
<!DOCTYPE html>
<html lang="{{ str_replace('_', '-', app()->getLocale()) }}">
<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">

    <!-- CSRF Token -->
    <meta name="csrf-token" content="{{ csrf_token() }}">

    <title>{{ config('app.name', 'Laravel') }}</title>

    <!-- Scripts -->
    <script src="{{ asset('js/app.js') }}" defer></script>

    <!-- Fonts -->
    <link rel="dns-prefetch" href="//fonts.gstatic.com">
    <link href="https://fonts.googleapis.com/css?family=Nunito" rel="stylesheet">

    <!-- File Manager CSS -->
    <link rel="stylesheet" href="{{ asset('vendor/file-manager/css/file-manager.css') }}">

    <!-- Styles -->
    <link href="{{ asset('css/app.css') }}" rel="stylesheet">

    <!-- Font Awesome 5 -->
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.7.0/css/all.css">
    
    <!-- Bootstrap 4 -->
    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.1.3/css/bootstrap.min.css">
   
    <!-- CSRF Token -->
    <meta name="csrf-token" content="{{ csrf_token() }}">
</head>
<body>
    <div id="app">
        <nav class="navbar navbar-expand-md navbar-light bg-white shadow-sm">
            <div class="container">
                <a class="navbar-brand" href="{{ url('/') }}">
                    {{ config('app.name', 'Laravel') }}
                </a>
                <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="{{ __('Toggle navigation') }}">
                    <span class="navbar-toggler-icon"></span>
                </button>

                <div class="collapse navbar-collapse" id="navbarSupportedContent">
                    <!-- Left Side Of Navbar -->
                    <ul class="navbar-nav mr-auto">

                    </ul>

                    <!-- Right Side Of Navbar -->
                    <ul class="navbar-nav ml-auto">
                        <!-- Authentication Links -->
                        @guest
                            <li class="nav-item">
                                <a class="nav-link" href="{{ route('login') }}">{{ __('Login') }}</a>
                            </li>
                            @if (Route::has('register'))
                                <li class="nav-item">
                                    <a class="nav-link" href="{{ route('register') }}">{{ __('Register') }}</a>
                                </li>
                            @endif
                        @else
                            <li class="nav-item dropdown">
                                <a id="navbarDropdown" class="nav-link dropdown-toggle" href="#" role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" v-pre>
                                    {{ Auth::user()->name }} <span class="caret"></span>
                                </a>

                                <div class="dropdown-menu dropdown-menu-right" aria-labelledby="navbarDropdown">
                                    <a class="dropdown-item" href="{{ route('logout') }}"
                                       onclick="event.preventDefault();
                                                     document.getElementById('logout-form').submit();">
                                        {{ __('Logout') }}
                                    </a>

                                    <form id="logout-form" action="{{ route('logout') }}" method="POST" style="display: none;">
                                        @csrf
                                    </form>
                                </div>
                            </li>
                        @endguest
                    </ul>
                </div>
            </div>
        </nav>

        <main class="py-4">
            @yield('content')
        </main>
    </div>

        <!-- File Manager JS -->
    <script src="{{ asset('vendor/file-manager/js/file-manager.js') }}"></script>

</body>
</html>

```
Here's screenshot of application where the loading icon is showing. 
![error](https://user-images.githubusercontent.com/23056780/66645353-dc530080-ec40-11e9-8e84-16766c512988.png)

I'm using XAMPP Version 7.3.9.

**Comment by alexusmai** [comment]
Hi,

You are using CSRF Token twice.

Open chrome console - do you have any errors?
Open tab - Network - check Request URL for initialize request - it should be something like this - http://localhost:8000/file-manager/initialize

![1](https://user-images.githubusercontent.com/7473505/66649661-71e99280-ec37-11e9-99f9-874588cf57d4.jpg)

**Comment by praful-dhabekar** [comment]
@alexusmai  Thanks for replying so fast but below screenshot is from my chrome's console tab. I don't have any error in the console:
![image](https://user-images.githubusercontent.com/23056780/66650045-4bceed00-ec4d-11e9-929b-0c722a49c915.png)


Below screenshot is from my chrome's network tab. I don't have any error in the network  tab:

![image](https://user-images.githubusercontent.com/23056780/66650521-74a3b200-ec4e-11e9-85e0-31b096419f1f.png)



I am only using CSRF token in **app.blade.php**

**Comment by alexusmai** [comment]
Yes, you are using CSRF-token in app.blade.php twice.

```
<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">

    <!-- CSRF Token -->
    <meta name="csrf-token" content="{{ csrf_token() }}">

    <title>{{ config('app.name', 'Laravel') }}</title>

    <!-- Scripts -->
    <script src="{{ asset('js/app.js') }}" defer></script>

    ....
    
    <!-- Bootstrap 4 -->
    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.1.3/css/bootstrap.min.css">
   
    <!-- CSRF Token -->
    <meta name="csrf-token" content="{{ csrf_token() }}">
</head>
```

Maybe problems with filesystem configuration.. please show me - config/filesystems.php - disks array - local disk settings

**Comment by praful-dhabekar** [comment]
I have removed extra CSRF token but still its not working. Just to check I have added following code to welcome.blade.php (on the welcome page) and its working 

![image](https://user-images.githubusercontent.com/23056780/66730050-c1fd6a80-ee6c-11e9-89f3-63a2b956f381.png)


```html
<div class="container">
            <div style="height: 700px;">
                <div id="fm"></div>
            </div>
 </div>
```
but if I add above code in home.blade.php which comes after login, giving me error. I can't upload or create folder.

Bellow is my config/filesystems.php

```
<?php

return [

    /*
    |--------------------------------------------------------------------------
    | Default Filesystem Disk
    |--------------------------------------------------------------------------
    |
    | Here you may specify the default filesystem disk that should be used
    | by the framework. The "local" disk, as well as a variety of cloud
    | based disks are available to your application. Just store away!
    |
    */

    'default' => env('FILESYSTEM_DRIVER', 'local'),

    /*
    |--------------------------------------------------------------------------
    | Default Cloud Filesystem Disk
    |--------------------------------------------------------------------------
    |
    | Many applications store files both locally and in the cloud. For this
    | reason, you may specify a default "cloud" driver here. This driver
    | will be bound as the Cloud disk implementation in the container.
    |
    */

    'cloud' => env('FILESYSTEM_CLOUD', 's3'),

    /*
    |--------------------------------------------------------------------------
    | Filesystem Disks
    |--------------------------------------------------------------------------
    |
    | Here you may configure as many filesystem "disks" as you wish, and you
    | may even configure multiple disks of the same driver. Defaults have
    | been setup for each driver as an example of the required options.
    |
    | Supported Drivers: "local", "ftp", "sftp", "s3", "rackspace"
    |
    */

    'disks' => [

        'local' => [
            'driver' => 'local',
            'root' => storage_path('app'),
        ],

        'public' => [
            'driver' => 'local',
            'root' => storage_path('app/public'),
            'url' => env('APP_URL').'/storage',
            'visibility' => 'public',
        ],

        's3' => [
            'driver' => 's3',
            'key' => env('AWS_ACCESS_KEY_ID'),
            'secret' => env('AWS_SECRET_ACCESS_KEY'),
            'region' => env('AWS_DEFAULT_REGION'),
            'bucket' => env('AWS_BUCKET'),
            'url' => env('AWS_URL'),
        ],

    ],

];


```

**Comment by alexusmai** [comment]
Did you create a symbolic link? [doc](https://laravel.com/docs/5.8/filesystem#the-public-disk)

**Comment by praful-dhabekar** [comment]
`php artisan storage:link`
Yes, I created a symbolic link. But still issue is coming.

**Comment by alexusmai** [comment]
Show me the results of these requests:

![1](https://user-images.githubusercontent.com/7473505/66855149-670e6500-ef8b-11e9-9a33-29505d19036c.jpg)
![2](https://user-images.githubusercontent.com/7473505/66855154-6970bf00-ef8b-11e9-9ca8-19fa73003091.jpg)
![3](https://user-images.githubusercontent.com/7473505/66855158-6bd31900-ef8b-11e9-84d3-00e09723443b.jpg)

**Comment by waleedbhatti** [comment]
![file-manager](https://user-images.githubusercontent.com/59745911/147408889-16baa222-f71e-49c2-9f0e-46b852ab174c.png)

Hello, I am adding a feature to my existing project. I've installed the package according to installation. I've attached the screenshot. I think there is an issue in the request. I cannot upload any file or make a directory. But have full preview. Kindly check and suggest. Any help would be appreciated.

Waiting for your reply,
Thanks
